unique_sdk/unique_sdk/_util.py

```diff
@@ -188,3 +192,56 @@ def _wrapper(*args, **kwargs):
                 return class_method(*args, **kwargs)
 
         return _wrapper
+
+
+def retry_on_error(
+    error_messages: List[str],
+    max_retries: int = 3,
+    initial_delay: int = 1,
+    backoff_factor: int = 2,
+    error_class=APIError,
+):
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        async def async_wrapper(*args, **kwargs) -> Any:
+            attempts = 0
+            while attempts < max_retries:
+                try:
+                    return await func(*args, **kwargs)
+                except Exception as e:
+                    if not any(err_msg in str(e) for err_msg in error_messages):
```

Comment:
if not any(err_msg.lower() in str(e).lower() for err_msg in error_messages):

you can include some normalization (e.g., stripping, lowercasing) if the errors might vary in format or case.

Reply:
Ah, this makes sense.